lib/matplotlib/colorbar.py

```diff
@@ -636,14 +604,16 @@ def _reset_locator_formatter_scale(self):
         the mappable normal gets changed: Colorbar.update_normal)
         """
         self.locator = None
+        self.minorlocator = None
         self.formatter = None
-        if isinstance(self.norm, colors.LogNorm):
-            # *both* axes are made log so that determining the
-            # mid point is easier.
-            self.ax.set_xscale('log')
-            self.ax.set_yscale('log')
-            self.minorticks_on()
-            self.__scale = 'log'
+        if hasattr(self.norm, '_scale'):
+            self.ax.set_xscale(self.norm._scale.name,
+                               **self.norm._scale._kwargs)
```

Comment:
Instead of storing `kwargs` to create another scale instance, perhaps you could instead modify `Axes.set_scale` (or add a new private `Axes._set_scale_instance`) that directly takes the scale instance as parameter instead?  The scale docs explicitly state that a single scale instance should be usable on multiple axes at the same time.

Reply:
Yeah, I wasn't brave enough to try that yet - we don't have, so far as I can tell, a public way to change the scale by just passing a scale instance to either the axes or axis.  Perhaps that needs to be added, though I could do it using private things here....

Reply:
Ahh, that was easier than I thought.  I just changed `axis._set_scale` to allow `ScaleBase` subclasses as well as strings.  Publicly that now means that `ax.set_xscale` will allow a `ScaleBase` as an argument.  I don't know why it didn't before.